laravel/jetstream
# Add initial components for Inertia React support

### **Title**: Add Full React Support for Inertia in Jetstream

### **Description**:
This PR introduces full support for React as a stack option for Jetstream when using Inertia. While Jetstream originally supported Vue with Inertia, this update brings React into the fold, allowing developers to leverage their preferred frontend framework within the Jetstream ecosystem.

Key changes:
- New installation commands:  
  - For Vue: `php artisan jetstream:install inertia-vue --teams`  
  - For React: `php artisan jetstream:install inertia-react --teams`
- All functionality previously available with Vue (authentication, teams, profile management, etc.) is now available with React, making the switch seamless for developers who prefer React for their applications.

### **Why the Community Needs This**:
The Laravel community is diverse, with many developers opting for React in their projects. By adding official support for React, Jetstream becomes more flexible and adaptable to a wider audience. This change will not only attract more React developers to use Jetstream but will also enhance the development experience for those already familiar with React.

### **Testing the PR**:
To test this PR locally:
1. Check out the PR and rename the package in your local environment (e.g., `my/jetstream`).
2. Update your `composer.json`:
    ```json
    "repositories": [
        {
            "type": "path",
            "url": "../path/to/this-checked-out-PR"
        }
    ]
    ```
3. In your project, run the following commands:
    - `laravel new test-app`
    - `composer require my/jetstream:@dev`
    - `php artisan jetstream:install inertia-react --teams`
4. Proceed with your usual development workflow.

This addition to Jetstream is an essential enhancement that aligns with the community’s growing demand for React support within Laravel applications. Looking forward to your feedback and contribution!
